### PR TITLE
created implementation for ShortestPathDAO

### PR DESCRIPTION
**Pull request by TomashevskiiVladimirSolvd**
please review it.

### CODE REVIEW

File: src/main/java/org/example/dao/ShortestPathDAOImpl.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+public class ShortestPathDAOImpl implements ShortestPathDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public ShortestPathDAOImpl() {
+    }
+
+    @Override
+    public void insertShortestPath(ShortestPath shortestPath) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
```
Comment (remynewton) [inline src/main/java/org/example/dao/ShortestPathDAOImpl.java]: I don't think the connection should be created in the try with resources. Use a finally block

**Review by remynewton — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/dao/ShortestPathDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public ShortestPath getShortestPath(int id) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
```
Comment (kabuki-mono) [inline src/main/java/org/example/dao/ShortestPathDAOImpl.java]: Missed one here

**Review by kabuki-mono — APPROVED** [review]
LGTM!

File: src/main/java/org/example/dao/ShortestPathDAOImpl.java (hunk 2 above)
Comment (TomashevskiiVladimirSolvd) [reply src/main/java/org/example/dao/ShortestPathDAOImpl.java]: one sec)

Comment (TomashevskiiVladimirSolvd) [reply src/main/java/org/example/dao/ShortestPathDAOImpl.java]: I have 2 updates,please check the last one.

**Review by remynewton — APPROVED** [review]
LGTM

File: src/main/java/org/example/dao/ShortestPathDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public ShortestPathDAOImpl() {
+    }
+
+    @Override
+    public void insertShortestPath(ShortestPath shortestPath) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (
+             PreparedStatement statement = connection.prepareStatement("INSERT INTO shortest_paths (id, previous_point, distance) VALUES (?, ?, ?)")) {
```
Comment (AndrewTchaikovsky) [inline src/main/java/org/example/dao/ShortestPathDAOImpl.java]: I can't view the schema right now but if there is an ID column here we should use generated keys and the while loop like in the other two DAO implementations.

**Review by AndrewTchaikovsky — DISMISSED** [review]

File: src/main/java/org/example/dao/ShortestPathDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public ShortestPath getShortestPath(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (
+             PreparedStatement statement = connection.prepareStatement("SELECT sp.id AS shortestPathId, sp.previous_point AS previousPoint, sp.distance AS distance FROM shortest_paths sp WHERE id = ?")) {
+            statement.setInt(1, id);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    PointDAO pointDAO = new PointDAOImpl();
+
+                    int shortestPathId = resultSet.getInt("shortestPathId");
```
Comment (AndrewTchaikovsky) [inline src/main/java/org/example/dao/ShortestPathDAOImpl.java:72]: The names in parenthesis should correspond to the names of the columns in the database not the names of variables in Java.

File: src/main/java/org/example/dao/ShortestPathDAOImpl.java (hunk 3 above)
Comment (TomashevskiiVladimirSolvd) [reply src/main/java/org/example/dao/ShortestPathDAOImpl.java]: will do

Comment (TomashevskiiVladimirSolvd) [reply src/main/java/org/example/dao/ShortestPathDAOImpl.java]: done

**Review by AndrewTchaikovsky — APPROVED** [review]

**Review by AndrewTchaikovsky — APPROVED** [review]

**Pull request merged**